### PR TITLE
Update rules_apple

### PR DESCRIPTION
**Pull request by vladmos**
The currently used version of rules_apple is incompatible with upcoming Bazel releases and will cause breakages.

release notes: no

@nicolasnoble

### CODE REVIEW

File: bazel/grpc_deps.bzl
Comment (gnossen) [inline bazel/grpc_deps.bzl:343]: This will require a manual push from a teammember; it's not a pull-through cache. You can just remove this line for the moment and I'll come back and add it in after your change is merged.

Comment (vladmos) [reply bazel/grpc_deps.bzl]: Done.

**Comment by gnossen** [comment]
@vladmos 

Looks like there's a checksum mismatch:

```
[https://github.com/bazelbuild/rules_apple/archive/709d8b12d68b571f23a1187b1cf7b10bb89fa46f.tar.gz] to /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/external/build_bazel_rules_apple/temp14091365186222698153/709d8b12d68b571f23a1187b1cf7b10bb89fa46f.tar.gz: Checksum was a0b7066d9f4193e8a800db2a66f4fe867a7806b8cdeecc3f0b9c9b3fc415338e but wanted c84962b64d9ae4472adfb01ec2cf1aa73cb2ee8308242add55fa7cc38602d882
```

**Comment by vladmos** [comment]
Thanks, updated the checksum.

File: bazel/grpc_deps.bzl
Comment (jtattermusch) [reply bazel/grpc_deps.bzl:343]: I just updated the mirror and added back the line into this PR.

**Review by jtattermusch — APPROVED** [review]
LGTM once the tests are green.

**Review by jtattermusch — CHANGES_REQUESTED** [review]
A bunch of tests that seem related to this change is failing:
https://source.cloud.google.com/results/invocations/b6c782c0-f9f1-476a-9331-43c4678a1d50/targets

```
if [ "$SCHEME" == "watchOS-sample-WatchKit-App" ]; then
  SCHEME="watchOS-sample watchOS-sample-watchApp"
fi
../../../tools/bazel build $SCHEME
INFO: Running bazel wrapper (see //tools/bazel for details), bazel version 3.7.1 will be used instead of system-wide bazel installation.
Loading:
Loading: 0 packages loaded
DEBUG: /Volumes/BuildData/tmpfs/tmp/bazel/external/bazel_toolchains/rules/rbe_repo/version_check.bzl:68:14:
Current running Bazel is ahead of bazel-toolchains repo. Please update your pin to bazel-toolchains repo in your WORKSPACE file.
DEBUG: /Volumes/BuildData/tmpfs/tmp/bazel/external/bazel_toolchains/rules/rbe_repo/checked_in.bzl:125:14: rbe_default not using checked in configs; Bazel version 3.7.1 was picked/selected but no checked in config was found in map {"0.20.0": ["8.0.0"], "0.21.0": ["8.0.0"], "0.22.0": ["8.0.0", "9.0.0"], "0.23.0": ["8.0.0", "9.0.0"], "0.23.1": ["8.0.0", "9.0.0"], "0.23.2": ["9.0.0"], "0.24.0": ["9.0.0"], "0.24.1": ["9.0.0"], "0.25.0": ["9.0.0"], "0.25.1": ["9.0.0"], "0.25.2": ["9.0.0"], "0.26.0": ["9.0.0"], "0.26.1": ["9.0.0"], "0.27.0": ["9.0.0"], "0.27.1": ["9.0.0"], "0.28.0": ["9.0.0"], "0.28.1": ["9.0.0"], "0.29.0": ["9.0.0"], "0.29.1": ["9.0.0", "10.0.0"], "1.0.0": ["9.0.0", "10.0.0"]}
DEBUG: /Volumes/BuildData/tmpfs/tmp/bazel/external/bazel_toolchains/rules/rbe_repo/version_check.bzl:68:14:
Current running Bazel is ahead of bazel-toolchains repo. Please update your pin to bazel-toolchains repo in your WORKSPACE file.
DEBUG: /Volumes/BuildData/tmpfs/tmp/bazel/external/bazel_toolchains/rules/rbe_repo/checked_in.bzl:125:14: rbe_msan not using checked in configs; Bazel version 3.7.1 was picked/selected but no checked in config was found in map {"0.20.0": ["8.0.0"], "0.21.0": ["8.0.0"], "0.22.0": ["8.0.0", "9.0.0"], "0.23.0": ["8.0.0", "9.0.0"], "0.23.1": ["8.0.0", "9.0.0"], "0.23.2": ["9.0.0"], "0.24.0": ["9.0.0"], "0.24.1": ["9.0.0"], "0.25.0": ["9.0.0"], "0.25.1": ["9.0.0"], "0.25.2": ["9.0.0"], "0.26.0": ["9.0.0"], "0.26.1": ["9.0.0"], "0.27.0": ["9.0.0"], "0.27.1": ["9.0.0"], "0.28.0": ["9.0.0"], "0.28.1": ["9.0.0"], "0.29.0": ["9.0.0"], "0.29.1": ["9.0.0", "10.0.0"], "1.0.0": ["9.0.0", "10.0.0"]}
Analyzing: target //src/objective-c/examples:Sample (0 packages loaded, 0 targets configured)
ERROR: /Volumes/BuildData/tmpfs/tmp/bazel/external/build_bazel_rules_apple/apple/internal/BUILD:547:18: in environment_plist rule @build_bazel_rules_apple//apple/internal:environment_plist_ios:
Traceback (most recent call last):
	File "/Volumes/BuildData/tmpfs/tmp/bazel/external/build_bazel_rules_apple/apple/internal/environment_plist.bzl", line 54, column 22, in _environment_plist
		apple_support.run(
	File "/Volumes/BuildData/tmpfs/tmp/bazel/external/build_bazel_apple_support/lib/apple_support.bzl", line 202, column 5, in _run
		def _run(ctx, xcode_path_resolve_level = _XCODE_PATH_RESOLVE_LEVEL.none, **kwargs):
Error: _run() missing 1 required positional argument: ctx
ERROR: Analysis of target '//src/objective-c/examples:Sample' failed; build aborted: Analysis of target '@build_bazel_rules_apple//apple/internal:environment_plist_ios' failed
INFO: Elapsed time: 0.243s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 26 targets configured)
FAILED: Build did NOT complete successfully (0 packages loaded, 26 targets configured)

2021-05-17 04:44:03,706 FAILED: ios-buildtest-example-sample [ret=1, pid=68705, time=0.3sec]
```

As such, I don't think we can merge this.

**Comment by jtattermusch** [comment]
I created https://github.com/grpc/grpc/pull/27410 which also updates rules_apple and it also updates to bazel 4, which hopefully fixes the problem reported above.

I'm closing this as superseded by https://github.com/grpc/grpc/pull/27410.

**Pull request closed without merging**
